compiler/rustc_attr_data_structures/src/lib.rs

```diff
@@ -11,6 +11,10 @@ mod stability;
 mod version;
 
 pub use attributes::*;
-pub(crate) use rustc_session::HashStableContext;
 pub use stability::*;
 pub use version::*;
+
+/// Requirements for a `StableHashingContext` to be used in this crate.
+/// This is a hack to allow using the `HashStable_Generic` derive macro
+/// instead of implementing everything in `rustc_middle`.
+pub trait HashStableContext: rustc_ast::HashStableContext + rustc_abi::HashStableContext {}
```

Comment:
this is the right way to do this. Importing the one from rustc_session isn't nice anyway and now also causes a circular dep